com.unity.shadergraph/Editor/Generation/Targets/BuiltIn/Editor/ShaderGraph/Includes/LegacySurfaceVertex.hlsl

```diff
@@ -0,0 +1,26 @@
+#ifndef UNITY_LEGACY_SURFACE_VERTEX
+#define UNITY_LEGACY_SURFACE_VERTEX
+
+struct v2f_surf {
+  float4 pos;//UNITY_POSITION(pos);
+  float3 worldNormal;// : TEXCOORD1;
+  float3 worldPos;// : TEXCOORD2;
+  float3 viewDir;
+  float4 lmap;// : TEXCOORD3;
+  #if UNITY_SHOULD_SAMPLE_SH
+  half3 sh;// : TEXCOORD3; // SH
+  #endif
+  float1 fogCoord; //UNITY_FOG_COORDS(4)
+  DECLARE_LIGHT_COORDS(4)//unityShadowCoord4 _LightCoord;
+  UNITY_SHADOW_COORDS(5)//unityShadowCoord4 _ShadowCoord;
+
+  //#ifdef DIRLIGHTMAP_COMBINED
+  float4 tSpace0 : TEXCOORD6;
```

Comment:
Wondering: v2f_surf struct for some members uses TEXCOORDn semantic, but for most it does not use anything (with a bunch of things commented out). Any reason for having semantic on some members?

Reply:
I believe this was just me forgetting about this as I was iterating. The built-in structs like v2f_surf are used to just pack and unpack data from shader graph before calling into the built-in functions. The semantics should therefore have no meaning. I believe I added the tSpaceN variables later and just didn't remove the semantics.